add extra column to users table

jackstar12 opened the pull request:
Adding an `attrs` JSON column to the users enables more use cases for third parties and other extensions.

For example, the discord extension will be able to store the associated `discord_id`, eliminating the need to basically cope all the crud code over.

dni commented on migrations.py:
```diff
@@ -29,3 +29,14 @@ async def m001_initial(db):
+    """
+    await db.execute(
+        """
+        ALTER TABLE usermanager.users ADD COLUMN attrs JSON
```
is JSON i actually db type? never saw this? does this work for postgres adn sqllite? and why not just TEXT?

jackstar12 replied:
yes it is, tried it out with sqlite aswell.
It's better than text because it allows for querying against json attributes with the ->> operator.

dni commented on crud.py:
```diff
@@ -22,10 +23,11 @@ async def create_usermanager_user(data: CreateUserData) -> User:
         """
-        INSERT INTO usermanager.users (id, name, admin, email, password)
-        VALUES (?, ?, ?, ?, ?)
+        INSERT INTO usermanager.users (id, name, admin, email, password, attrs)
```
i would call it `extra` instead of attrs. we use extra on our payment aswell

jackstar12 replied:
makes sense. 
should probably add an endpoint for the method aswell something like 
GET /api/v1/usermanager/user?admin=23452345&extra.discord_id=6734572347

jackstar12 commented:
@dni added an advanced filtering endpoint.
The `Filter`  class is ported from another project of mine and can be used in other places aswell (might add it to the helpers in core?)

jackstar12 commented:
@dni added openapi docs and refactored paths, removing some redundant code.
I introduced a `UserDetailed` model which includes the wallets. Its being returned on when querying a specific user or creating one.
I think it's ready to be merged.

I've generally found the approach of documenting the API inside the extension page to be rather weird aswell, because openapi docs are way easier to create and try out for the end user
